lib/sublayer/components/output_adapters/formattable.rb

```diff
@@ -0,0 +1,29 @@
+module Sublayer
+  module Components
+    module OutputAdapters
+      module Formattable
+        def format_properties
+          formatted_properties = {}
+          self.properties.each do |prop|
+            property = {
+              type: prop.type,
+              description: prop.description
+            }
+
+            property[:enum] = prop.enum if prop.respond_to?(:enum) && prop.enum
+            property[:default] = prop.default if prop.respond_to?(:default) && !prop.default.nil?
+            property[:minimum] = prop.minimum if prop.respond_to?(:minimum) && !prop.minimum.nil?
+            property[:maximum] = prop.maximum if prop.respond_to?(:maximum) && !prop.maximum.nil?
```

Comment:
Do any of the AI providers use minimum/maximum/default in their specs? Probably not super important right now I guess...and probably more important to be json_spec compliant?

Reply:
yea it seems like its just absorbed that way by the llm as well (learned this from the spec error in openai but it not causing issues) 

it's probably just passed along to be digested by the llm that is already trained on the json schema spec. 

i dont know if i saw any examples of default/min/max but yea i figured its an okay start and we'll keep adding more as we add more output adapters. there are going to be a lot...